policy-format.md

```diff
@@ -0,0 +1,172 @@
+# Explainer: Format of Origin Policy documents
+
+**tl;dr** [Origin Policy](README.md) documents are JSON files with top-level
+entries for each supported policy mechanism.
+
+## General File Format
+
+An origin policy document is a JSON document that contains a series of policy
+items. A top-level dictionary names each policy items and contains item specifc
+calues. Each type of (currently) supported policy item is described below.
+
+Example:
+
+```js
+{
+  "tls": .... ,
+  "content-security-policy": .... ,
+  "referrer": ....
+}
+```
+
+## Format Versioning and Error Handling
+
+Applications should
+
+- raise an error if the file is not well-formed JSON,
+- ignore any top-level section they do not understand,
+- consider it an error if they fail to parse a policy item that they do
+  understand.
+
+**TODO:** Should there be a 'must understand' directive, so that site owners
+can determine themselves whether failure to understand a given policy item
+should be fatal ot nor?
+
+## Support Policy Items
+
+### Content Security Policy (CSP)
+
+The `content-security-policy` policy item contains the equivalent of one or more
+[Content-Security-Policy HTTP headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP).
+
+- The `"content-security-policy"` policy item contains an array of
+  content-security-policy directives.
+- Each CSP directive contains a `"policy"` string, containing the CSP format.
+- A `"disposition"` is either `"enforce"` or `"report"`.
+
+Example:
+
+```js
+"content-security-policy": [
```

Comment:
If we want to support "must-understand", every top-level thing will need to be a dictionary.

Reply:
> [where "must-understand" should really just be the top-level.]

Huh. That's not at all the way I'd do it. :)

In my opinion, the top-level keys should be the things that are being controlled. That's partially just a style preference that I will blindly assert is actually easier to read by humans (:fire:), and partially because I want this mechanism to be able to interop with application manifests in the same file (see https://wicg.github.io/origin-policy/#app-manifest).

> There is a related question: It strikes me as better and easier if repeated directives are just repeated top-level keys. That is, instead of `"csp": [{...}, {..}]` you'd do `"csp": {...}, "csp": {....}`.
That feels natural for CSP, but not so for some other properties.

Note the second paragraph in https://tools.ietf.org/html/rfc7159#section-4. Repeated keys are not interoperable. I'd very much prefer that we do not rely on them (also because, as you note, that's not how you'd expect a dictionary to work :) ).

Reply:
Done. (Changed all formats to be top-level dictionaries, and established this as a general rule.)